package.json

```diff
@@ -32,6 +33,9 @@
     "prettier-check": "^2.0.0",
     "webpack": "^4.39.3"
   },
+  "peerDependencies": {
+    "@types/webpack": "^4.0.0 || ^5.0.0"
+  },
```

Comment:
I'm not sure about this. This way we will make everyone install it, even when they are not using TypeScript. Otherwise, we'll warn every time they don't.

Reply:
Warning is fine IMO. But, when does one install the webpack plugin and not webpack?

Reply:
Never, but those are types, not webpack itself.

Reply:
Yeah, I debated about that. I can't believe we put a man on the moon _more than half a century ago_ and still haven't solved this problem. Either we:

- make it a hard dependency ( which is what the TS docs [would have us do](https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html#dependencies)), and plain JS people install a package they don't need, or

- make it a peer dependency, and plain JS people get warnings which don't apply to them, or

- make it neither, and TS people get yelled at by their linter and/or compiler unless they happen to have the relevant types installed.

Which is the least bad, do we think?

Reply:
The first one is a no-go for me, and I rather the third over the second although not strongly opinionated against the second one. 

- Why should JS people get warnings related to TS?
- TS people know they sometimes need to install additional modules for the types, so this shouldn't be a big deal.

Reply:
I'd also go with 3rd. If you are using TypeScript *and* Webpack, there's 99,99% chance that you already have those types installed.

Reply:
> TS people know they sometimes need to install additional modules for the types, so this shouldn't be a big deal.

Yeah, but that's generally for their _own_ dependencies, not transitive ones. I agree it's not the world's biggest blocker, and people will figure it out, so in this case I'm happy to go with option 3, especially since you're right, @kamilogorek, it's not like webpack is obscure and at least some folks will already have the types. (I'm less convinced it's everyone, though. If you're using our nextjs SDK and don't have any custom webpack config other than what you pass to our plugin, there's no reason you'd already have them.) So I can pull it.

As a general principle, though, and given just how popular TS is, it's still just amazing to me that npm/yarn haven't adapted to what has to be a pretty common situation/problem.

Reply:
> As a general principle, though, and given just how popular TS is, it's still just amazing to me that npm/yarn haven't adapted to what has to be a pretty common situation/problem.

Not sure this is only an npm/yarn thing, but also from the npm registry? But yes, there should be a solution to this.

Reply:
> Not sure this is only an npm/yarn thing, but also from the npm registry? 

Not sure what you mean.

Reply:
The npm registry is [npmjs.com](https://www.npmjs.com/), and then the `npm` tool which you use to manage the modules in the registry. Anyways, I haven't dived into the registry and I don't know where the solution should be coming from, just giving my thoughts.